components/service/experiments/src/test/java/mozilla/components/service/experiments/ExperimentTest.kt

```diff
@@ -14,33 +14,33 @@ import org.robolectric.RobolectricTestRunner
 @RunWith(RobolectricTestRunner::class)
 class ExperimentTest {
     @Test
-    fun testEquals() {
-        val experiment = Experiment(
-            "id",
-            "name",
-            "description",
-            null,
-            null,
-            12345,
-            null)
+    fun `test equals()`() {
+        val experiment = testDefaultExperiment(
+            id = "id",
+            description = "description",
+            buckets = Experiment.Buckets(0, 100),
+            branches = emptyList(),
+            match = testDefaultMatcher(),
+            lastModified = 12345,
+            schemaModified = null)
+
         assertTrue(experiment == experiment)
         assertFalse(experiment.equals(null))
```

Comment:
nit: with `.equals()` implemented, you could use `==` instead of the call to `.equals(null)`.  Something like this:
```Kotlin
assertFalse(experiment == null)
```

Reply:
That gives IDE warning/highlighting on this never being true.

Reply:
Then maybe the `assertFalse` is unnecessary?  Is it possible for `testDefaultExperiment()` to return null?